File: Java/src/e2010_07/Task3.java
package e2010_07;

public class Task3 {

	public static void printPC(PC pc) {
		System.out.println("CPU: " + pc.getCores() + " cores of " + pc.getCPU() + "GHz");
		System.out.println("Storage: " + pc.getStorage());
		System.out.println("Network Access: " + pc.getNetworkAccess());
		System.out.println("Power: " + pc.getPower());
		System.out.println("-------------");
	}

	public static void printWifiConnectedDevices() {
		System.out.println(">>> WiFi Connected Devices: " + PC.WIFI_CONNECTED_DEVICES);
	}

	public static void main(String[] args) {

		printWifiConnectedDevices();

		PC defaultPC = new PC();
		printPC(defaultPC);
		printWifiConnectedDevices();

		PC customPC = new PC(1, 2.7, 512, true, true);
		printPC(customPC);
		printWifiConnectedDevices();

		PC copyPC = new PC(customPC);
		printWifiConnectedDevices();

		copyPC.setPower(false);

		printPC(copyPC);
		printWifiConnectedDevices();

	}

}

File: CPP/2010-07/task3.cc
#include <iostream>

using namespace std;

class PC {

private:
    int cores;
    float cpu;
    int storage;
    bool network_access;
    bool power;

public:
    static int WIFI_NETWORK_CONNECTIONS;

    PC() {}

    PC(int cores, float cpu, int storage, bool network_access, bool power) {
        this->cores = cores;
        this->cpu = cpu;
        this->storage = storage;
        this->network_access = network_access;
        this->power = power;

        if (network_access) {
            WIFI_NETWORK_CONNECTIONS++;
        }
    }

    PC(const PC& pc) {
        this->cpu = pc.cpu;
        this->cores = pc.cores;
        this->storage = pc.storage;
        this->network_access = pc.network_access;
        this->power = pc.power;

        if (pc.network_access) {
            WIFI_NETWORK_CONNECTIONS++;
        }
    }

    float get_cpu() {
        return cpu;
    }

    float get_cores() {
        return cores;
    }

    void set_cores(int cores) {
        this->cores = cores;
    }

    int get_storage() {
        return this->storage;
    }

    bool get_network_access() {
        return this->network_access;
    }

    bool get_power() {
        return this->power;
    }

    void set_power(bool power) {
        this->power = power;
    }
};

int PC::WIFI_NETWORK_CONNECTIONS;

void print_pc(PC* pc) {
    cout << "CPU: " << pc->get_cores() << " cores of " << pc->get_cpu() << "GHz" << endl;
    cout << "Storage: " << pc->get_storage() << endl;
    cout << "Network Access: " << pc->get_network_access() << endl;
    cout << "Power: " << pc->get_power() << endl;
    cout << "-------------" << endl;
}

void print_wifi_connections() {
    cout << ">>> WiFi Connections: " << PC::WIFI_NETWORK_CONNECTIONS << endl;
}

int main(int argc, char** argv) {

    print_wifi_connections();

    PC default_pc = PC();
    print_pc(&default_pc);
    print_wifi_connections();

    PC custom_pc = PC(1, 2.4, 512, true, false);
    print_pc(&custom_pc);
    print_wifi_connections();

    PC copy_pc = custom_pc;

    copy_pc.set_cores(2);
    copy_pc.set_power(true);

    print_pc(&copy_pc);
    print_wifi_connections();

    return 0;
}
